src/macros-by-example.md

```diff
@@ -140,6 +140,12 @@ the syntax element that matched them. The keyword metavariable `$crate` can be
 used to refer to the current crate; see [Hygiene] below. Metavariables can be
 transcribed more than once or not at all.
 
+For reasons of backwards compatibility, although `_` is a (keyword) identifier,
+it is not matched by the `ident` fragment specifier. Similarly, though `_` [is
```

Comment:
I don't think this needs to mention `_` as an identifier.  The Reference doesn't define `_` as an [identifier](https://doc.rust-lang.org/nightly/reference/identifiers.html).  It's just punctuation. I think this may be a little different from rustc which treats it as a reserved identifier. But from a language perspective it was easier to define it as just punctuation.  I wouldn't want this statement to imply that `_` is an identifier.

This also reminds me, can you also add `expressions/underscore-expr.md` to the `_` entry in the [Punctuations table](https://github.com/rust-lang/reference/blob/master/src/tokens.md#punctuation)?